File: Math/1342. Number of Steps to Reduce a Number to Zero/1342. Number of Steps to Reduce a Number to Zero.py
class Solution:
    def numberOfSteps(self, num: int) -> int:

        steps = 0 #Acumulator variable
        
        #Loop for executing the steps to reduce to 0
        while num > 0:
            # Conditional, if num is even we split into 2 else we substract 1 from it
            if num % 2 == 0:
                num /= 2
            else:
                num -= 1
            steps += 1
        
        return steps
        
    

File: Math/1342. Number of Steps to Reduce a Number to Zero/1342. Number of Steps to Reduce a Number to Zero.ts
function numberOfSteps(num: number): number {
    var return_num: number = 0;
    while(num > 0){
        if(num % 2 == 0)
            num /= 2;
        else
            num -= 1;
        return_num ++;
    }
    return return_num;
};